nixos/modules/services/web-apps/privatebin.nix

```
@@ -0,0 +1,278 @@
+{ lib
+, pkgs
+, config
+, ...
+}:
+with lib;
+
+let
+  user = "privatebin";
+  group = user;
+  cfg = config.services.privatebin;
+  envVar = mkOptionType {
+    name = "environment variable";
+    description = "attribute set with _env as string";
+    descriptionClass = "composite";
+    check = x: isAttrs x && x ? _env && types.nonEmptyStr.check x._env;
+  };
+  isEnv = v: isAttrs v && v ? _env && isString v._env;
+  format =
+    let
+      iniAtom = (pkgs.formats.ini {}).type/*attrsOf*/.functor.wrapped/*attrsOf*/.functor.wrapped;
+    in
+    pkgs.formats.ini { } // {
+      type = with types; attrsOf (attrsOf (either iniAtom envVar));
+    };
+
+  expire_options = finalCfg.expire_options or {};
+  expire_lines = let
+    toList = mapAttrsToList (key: value: { inherit key value; });
+    sortExpire = builtins.sort (a: b: (a.value < b.value) && a.value != 0);
+  in
+    map (v: "${v.key}=${toString v.value}") (sortExpire (toList expire_options));
+
+  toIni = generators.toINI {
+    mkKeyValue = flip generators.mkKeyValueDefault "=" rec {
+      mkValueString = v:
+        if isString v then ''"${toString v}"''
+        else if isEnv v then "\${${v._env}}"
+        else generators.mkValueStringDefault { } v;
+    };
+  };
+
+  configFile = pkgs.writeTextDir "conf.php" ''
+    ;<?php http_response_code(403); /*
+    ${toIni (builtins.removeAttrs finalCfg ["expire_options"])}
+    ${optionalString (expire_options != {}) ''
+    [expire_options]
+    ${concatStringsSep "\n" expire_lines}
+    ''}
+    ;*/
+  '';
+
+  autoDb = if !cfg.databaseSetup.enable then null else cfg.databaseSetup.kind;
+  db_config = optionalAttrs (autoDb != null) (
+    if autoDb == "mysql" then {
+      model.class = "Database";
+      model_options = {
+        dsn = "mysql:unix_socket=/run/mysqld/mysqld.sock;dbname=${user}";
+        usr = user;
+      };
+    } else if autoDb == "pgsql" then {
+      model.class = "Database";
+      model_options.dsn = "pgsql:host=/run/postgresql;port=${toString config.services.postgresql.port};dbname=${user}";
+    } else if autoDb == "sqlite" then {
+      model.class = "Database";
+      model_options.dsn = "sqlite:/var/lib/privatebin/data/db.sqlite3";
+    } else {}
```

```suggestion
    };
```
there is no other option